### ISSUE TITLE
Forgetting Transformer: Model and Training Code Coming Soon?

### ISSUE THREAD

**Issue by NielsRogge** [zhixuan-lin/forgetting-transformer#1]
Hi @zhixuan-lin 🤗

I'm Niels from the Hugging Face open-source team. I came across your paper, "Forgetting Transformer: Softmax Attention with a Forget Gate," on arXiv and wanted to reach out. We're excited about your work on the Forgetting Attention mechanism and the FoX model!

We're building a new feature called Paper Pages ([hf.co/papers](https://hf.co/papers)) that connects arXiv papers with their associated code and datasets on the Hugging Face Hub. This makes research more discoverable and allows for easier reproducibility. We noticed your GitHub repository ([https://github.com/zhixuan-lin/forgetting-transformer](https://github.com/zhixuan-lin/forgetting-transformer)) and are particularly interested in helping you make your pre-trained models and datasets available on the Hugging Face Hub once they are ready.

Your README mentions that training and evaluation code is "Work in progress." Would you be open to sharing your models and datasets when available, and potentially collaborating with us to create a great model card when you release the training code? We can help with the metadata, tagging, and linking to the Paper Page.

Let us know if you're interested or have any questions!

Best regards,

Niels
ML Engineer @ Hugging Face 🤗

**Comment by zhixuan-lin** [comment]
Hi Niels,

Thanks for your interest! I plan to release the weights on the Hugging Face Hub later (maybe this weekend). I will let you know if I encounter any issues or have questions when I do that.

Our dataset is [LongCrawl64](https://manifestai.com/articles/longcrawl64/), which is publicly available and hosted by [Manifest AI](https://manifestai.com/), so I think there should be no need for me to upload a copy to the hub.

By the way, the training code is already in the repo. I just need to run some tests and update the instructions (again, maybe this weekend :)

**Comment by NielsRogge** [comment]
Ok great! Perhaps it'd be useful for the community to make the dataset available on the hub, we have features like the [Dataset viewer](https://huggingface.co/docs/hub/en/datasets-viewer) which makes it possible for people to view the first few rows in the browser, run SQL queries on them etc.

In case you make it compatible with the Datasets library, people could train open-source LLMs on them.

See here for a guide: https://huggingface.co/docs/datasets/loading.

**Comment by zhixuan-lin** [comment]
Hi Niels,

All our models have been uploaded to [this Hugging Face collection](https://huggingface.co/collections/zhixuan-lin/forgetting-transformer-paper-checkpoints-67d0ded3caa418ff0cc16ba4) and linked to this paper. As for the LongCrawl64 dataset, it was stored as [Zarr arrays](https://github.com/zarr-developers/zarr-python), which is not supported by HF by default.  I know it is technically possible to write a loading script for it (to convert it to HF format), but our code relies on the Zarr array layout for efficient dataloading. The dataset is also pre-tokenized without an official tokenizer, which would limit its applicability. For the above reasons, I don't plan to upload it to Hugging Face (it is publicly available and hosted by [Manifest AI](https://manifestai.com/) though, in case anyone wants to use it). 

Thank you for all your suggestions! Hugging Face is indeed a great place to share paper artifacts and super convenient to use :)
